include/zephyr/drivers/led.h

```diff
@@ -308,6 +331,15 @@ static inline int z_impl_led_on(const struct device *dev, uint32_t led)
 	const struct led_driver_api *api =
 		(const struct led_driver_api *)dev->api;
 
+	if (api->set_brightness == NULL &&
+	    api->on == NULL && api->off == NULL) {
+		return -ENOSYS;
+	}
+
+	if (api->on == NULL && api->off == NULL) {
```

Comment:
`api->off` should not matter here

Reply:
Oh, yeah, good point, dropped from here and the check above as well.